keps/sig-storage/3762-persistent-volume-last-phase-transition-time/README.md

```diff
@@ -0,0 +1,879 @@
+With `Retain` policy all volumes that are retained and left unclaimed have their phase is set to `Released`.
+As the released volumes pile up over time admins want to perform manual cleanup based on the time when the volume was
+last used, which is when the volume transitioned to `Released` phase.
+
+We can approach the solution in a more generic way and record a timestamp of when the volume transitioned to any phase,
+not just to `Released` phase. This allows anyone, incl. our perf tests, to measure time e.g. between a PV `Pending` and `Bound`
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
+
+1) Introduce a new status field in PersistentVolumes.
+2) Update the new field with a timestamp every time a volume transitions to a different phase (`pv.Status.Phase`).
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+### Non-Goals
+
+1) Implement any form of volume health monitoring.
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+We need to update API server to support the newly proposed field and update PV controller to set a value of the new
+timestamp field when a volume transitions to a different phase. Also, if the feature gate is disabled the value must be
+re-set to `nil` when updating or creating a volume.
+
+The value of the field is not intended for use by any other Kubernetes components at this point and should be used only
+as a convenience feature for cluster admins. Cluster admins should be able to list and sort PersistentVolumes based on
+a timestamp which indicates when the volume transitioned to a different state. 
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+As a cluster admin I want to use `Retain` policy for released volumes, which is safer than `Delete`, and implement a reliable policy to delete volumes that are `Released` for more than X days.
```

Comment:
Another story could be for cluster admin to delete a PVC if it has been pending for X hours.

Reply:
or alert

Reply:
We're not proposing PVC timestamps in this KEP. Should we?

Reply:
That's a good story - adding. See if you like the wording I put here.